stackhpc_monasca_agent_plugins/tests/unit/checks/test_slurm.py

```diff
@@ -77,6 +97,49 @@ def test__extract_node_names_series(self):
         actual = self.slurm._extract_node_names(field)
         self.assertEqual(expected, actual)
 
+    def test__extract_node_names_single_node(self):
+        field = "openhpc-compute-17"
+        expected = {
+            'openhpc-compute-17'
+        }
+        actual = self.slurm._extract_node_names(field)
+        self.assertEqual(expected, actual)
+
+    def test__extract_node_names_discontinuous(self):
+        field = "openhpc-compute-[1,15]"
+        expected = {
+            'openhpc-compute-1',
+            'openhpc-compute-15',
+        }
+        actual = self.slurm._extract_node_names(field)
+        self.assertEqual(expected, actual)
+
+    def test__extract_node_names_series_discontinuous(self):
+        field = "openhpc-compute-[1-3,5]"
+        expected = {
+            'openhpc-compute-1',
+            'openhpc-compute-2',
+            'openhpc-compute-3',
+            'openhpc-compute-5',
+        }
+        actual = self.slurm._extract_node_names(field)
+        self.assertEqual(expected, actual)
+
+    def test__extract_node_names_series_multiple_discontinuous(self):
+        field = "openhpc-compute-[1-3,5,7-9,11]"
```

Comment:
Have you seen this syntax used by slurm?

We can't handle multiple non-numeric prefixes currently. I think that's ok for our environment, although they might be named differently if we had some high memory or GPU nodes in the cluster, something like:
```
openhpc-compute-[0-15]
openhpc-himem-[0-1]
openhpc-gpu-[0-1]
```
Maybe just needs a comment somewhere about what this can and can't handle.

Reply:
Thanks - i've extrapolated from the format described in the docs: 
```https://github.com/SchedMD/slurm/blob/b2fadc636facb01af0100b0c4a44e162
4d0286fd/doc/html/api.shtml#L307```

I'll add a note about the limitations.

Reply:
Issued created to track multiple prefixes: https://github.com/stackhpc/stackhpc-monasca-agent-plugins/issues/4

Reply:
Ok. I believe the syntax with multiple prefixes is:

`openhpc-compute-[0-15],openhpc-himem-[0-1],openhpc-gpu-[0-1]`

We would need to verify this though. It should be possible to modify slurm.conf on the control node and see what it says.